src/workflow_tasks.rs

```diff
@@ -554,11 +604,26 @@ impl WorkflowTaskManager {
         }
     }
 
+    /// Must be called after *every* activation is replied to, regardless of whether or not we
+    /// had some issue reporting it to server or anything else. This upholds the invariant that
+    /// every activation we issue to lang has exactly one reply.
+    ///
+    /// Any subsequent action that needs to be taken will be created as a new activation
+    pub(crate) fn activation_done(&self, run_id: &str) {
+        if self.outstanding_activations.remove(run_id).is_some() {
```

Comment:
Would it be considered an error if there was nothing removed from `outstanding_activations`?
Also, consider renaming this method to something that clarifies it's a side-effect and doesn't just return a boolean. Maybe `on_activation_done` or `complete_activation`?

Reply:
Not an error b/c sometimes it's already been removed due to evicting. Added comment.